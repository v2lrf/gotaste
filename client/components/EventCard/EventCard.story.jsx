import React from 'react'
import { storiesOf } from '@storybook/react'

import EventCard from '.'

const fakeEvent = {
  name: 'Blindsmagningsaften',
  beginsAt: '2019-03-07T18:15:00Z',
  slug: '2018-03-07-blindsmagningsaften',
  eventHeroImageId: '5236SxUSokqXQeUdmD9gQFgR',
  host: {
    logoId: 'GoTaste/logos/Wh1V3cNhbCPrv3ZLXcxcgyWJ',
    name: 'Erik Sørensen Vin',
    address: {
      streetName: 'Store Kongensgade',
      streetNumber: '124',
      postalCode: '1264',
      city: 'København K'
    }
  }
}

storiesOf('EventCard', module).add('Default', () => (
  <div className="p-16">
    <EventCard {...fakeEvent} />
  </div>
))
