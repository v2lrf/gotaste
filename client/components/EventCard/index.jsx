import React from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import moment from 'moment'
import { Image } from 'cloudinary-react'

import config from '../../config'

import DateDisplay from '../DateDisplay'
import InfoItem from '../InfoItem'

function EventCard({ name, beginsAt, slug, host }) {
  const {
    logoId,
    name: hostName,
    address: { streetName, streetNumber, postalCode, city }
  } = host
  return (
    <Link href={{ pathname: '/event', query: { slug } }} as={`/event/${slug}`}>
      <a className="inline-flex flex-col rounded shadow no-underline hover:shadow-md mb-4">
        <div className="relative">
          <Image
            cloudName={config.cloudinaryCloudName}
            publicId="Govinu/hero_images/6c2543d032ac10028fc849e2b97cb1f7"
            height={200}
            width={600}
            crop="fill"
            className="rounded-t"
            secure="true"
          />
          <DateDisplay
            timestamp={beginsAt}
            className="absolute pin-t pin-l ml-2 mt-2"
          />
        </div>
        <div className="p-4">
          <time dateTime={beginsAt} className="text-grey-darker text-sm">
            {moment(beginsAt)
              .locale('da')
              .format('LLLL')}
          </time>
          <h2 className="mb-4 text-red-darker text-lg">{name}</h2>
          <InfoItem
            imageSrc={logoId}
            headline={hostName}
            tagline={`${streetName} ${streetNumber}, ${postalCode} ${city}`}
          />
        </div>
      </a>
    </Link>
  )
}

EventCard.propTypes = {
  name: PropTypes.string.isRequired,
  beginsAt: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  host: PropTypes.shape({
    logoId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    address: PropTypes.shape({
      streetName: PropTypes.string.isRequired,
      streetNumber: PropTypes.string.isRequired,
      postalCode: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
}

export default EventCard
