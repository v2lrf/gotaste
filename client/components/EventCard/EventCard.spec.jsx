import React from 'react'
import moment from 'moment-timezone'
import EventCard from '.'

moment.tz.setDefault('Europe/Copenhagen')

describe('EventCard', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      name: 'Blindsmagningsaften',
      beginsAt: '2019-03-07T18:15:00Z',
      slug: '2018-03-07-blindsmagningsaften',
      host: {
        logoId: 'Govinu/logos/Wh1V3cNhbCPrv3ZLXcxcgyWJ',
        name: 'Erik Sørensen Vin',
        address: {
          streetName: 'Store Kongensgade',
          streetNumber: '124',
          postalCode: '1264',
          city: 'København K'
        }
      }
    }
    component = shallow(<EventCard {...props} />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
