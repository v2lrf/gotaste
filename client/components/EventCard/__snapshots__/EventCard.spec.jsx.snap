// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Snapshots should match the snapshot 1`] = `
<Link
  as="/event/2018-03-07-blindsmagningsaften"
  href={
    Object {
      "pathname": "/event",
      "query": Object {
        "slug": "2018-03-07-blindsmagningsaften",
      },
    }
  }
>
  <a
    className="inline-flex flex-col rounded shadow no-underline hover:shadow-md mb-4"
  >
    <div
      className="relative"
    >
      <Image
        className="rounded-t"
        cloudName="dkrjpli3y"
        crop="fill"
        height={200}
        publicId="Govinu/hero_images/6c2543d032ac10028fc849e2b97cb1f7"
        secure="true"
        width={600}
      />
      <DateDisplay
        className="absolute pin-t pin-l ml-2 mt-2"
        timestamp="2019-03-07T16:15:00Z"
      />
    </div>
    <div
      className="p-4"
    >
      <time
        className="text-grey-darker text-sm"
        dateTime="2019-03-07T16:15:00Z"
      >
        torsdag d. 7. marts 2019 kl. 17:15
      </time>
      <h2
        className="mb-4 text-red-darker text-lg"
      >
        Blindsmagningsaften
      </h2>
      <InfoItem
        headline="Erik Sørensen Vin"
        imageAlt=""
        imageSrc="Govinu/logos/Wh1V3cNhbCPrv3ZLXcxcgyWJ"
        large={false}
        tagline="Store Kongensgade 124, 1264 København K"
      />
    </div>
  </a>
</Link>
`;
